client/src/pages/TeacherView.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { useLocation} from "react-router-dom";
+import "./teacher.css"
+
+//got an error due to useNavigate. To correct it: npm install react-router-dom@6
+
+
+function TeacherView() {
+    // State variables to manage component state
+    const [selectedCourse, setSelectedCourse] = useState(null); // Stores the currently selected course
+    const [courses, setCourses] = useState([]); // Stores the list of courses fetched from the API
+    const [isHovered, setIsHovered] = useState({}); // Tracks which course is being hovered
```

Comment:
good way for now, but eventually you would want to have component for this behaviour (i.e you have a parent component containing the list of courses and each element gets rendered into its own Course component)